memcached-image/memcached-image.xsl

```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+<xsl:output method="xml" encoding="UTF-8" indent="yes"/>
+<xsl:template match="/">
+<xsl:comment> OBS-ExclusiveArch: aarch64 x86_64 </xsl:comment>
+<xsl:comment>
+  This file is generated,
+  don't manually modify this file,
+  please check README
+</xsl:comment>
+
+<image schemaversion="6.9" name="memcached-image" xmlns:suse_label_helper="com.suse.label_helper">
+
+  <description type="system">
+    <author>Thomas Bechtold</author>
+    <contact>tbechtold@suse.com</contact>
+    <specification>memcached container</specification>
+  </description>
+
+  <preferences>
```

Comment:
Using https://osinside.github.io/kiwi/building/build_with_profiles.html#building-with-the-open-build-service you could put multiple profiles into one OBS package, so that the same package build works for multiple base distributions.

Reply:
Each profile will use one prefrences element.

Reply:
Reading the answers from https://lists.opensuse.org/opensuse-factory/2019-08/msg00086.html , using profiles does not work (or is at least not a good idea)

Reply:
One of the answer is that using different profiles is not possible for to use different OS, without saying how it won't work. The other one says that they have a feeling that it is hard, because having the same set of dependencies for two distros is hard (which we already do here).

So I don't yet understand why using multiple profiles will be more difficult.

But lets look at the problem instead of one solution: Do you already have an idea how to deal with copying the package (that builds the container) from OBS to IBS and where to build with which base OS?

Reply:
hm. Not sure if I get the question

Reply:
We'll need a memcached image based on SLES and one based on Leap.
Will both exist in OBS and IBS?
In which OBS/IBS project?
Which OBS/IBS package name will each have?
If both images are not built in both IBS and OBS, how will the build be disabled?

Reply:
> We'll need a memcached image based on SLES and one based on Leap.
> Will both exist in OBS and IBS?

I think we don't need the leap version in IBS. Not sure if there is a SLE base container the we can refer to on OBS.

> In which OBS/IBS project?

Why is that relevant here? We use a project and could create subprojects per distro (eg Cloud:OpenStack:Rocky:Containers:openSUSE_Leap_15.1) 

> Which OBS/IBS package name will each have?

I think always the same name (independent from the distro it builds for)

> If both images are not built in both IBS and OBS, how will the build be disabled?

If we use a subproject per distro, we just do not create the subproject.


I'm not sure if that (what I describe here) is the best solution. But it's a start. I'm happy to iterate over it and also try out other solutions.

Reply:
There is a SLE15 SP1 container in obs.

Reply:
I think we should avoid using subprojects and keep the whole product in one project. Otherwise we make many things including any staging process more complicated.

Reply:
Agreed. But the whole product is already split over multiple repos (eg. Cloud, SLE, SES, CaaSP).